### ISSUE TITLE
Exposing utilities ('classnames', lodash'...) to avoid duplicating them in third party blocks

### ISSUE THREAD

**Issue by marcusig** [WordPress/gutenberg#8078]
Would it be a good idea to expose some of the regularly used (and useful) packages like `classnames` and ~~`lodash`~~, in  - for example - `wp.utils` to allow third party plugins to use them? 

It would save having to include them in every plugin that creates blocks, while it's already part of Gutenberg. 

**Edit:** I just realised that lodash is already available. Is classnames also available somewhere?

**Comment by chrisvanpatten** [comment]
Was just talking about this with @dimofte — it'd be great to see `classnames` available as an external, like `lodash` is.

**Comment by aduth** [comment]
There's a certain amount of support commitment which comes with exposing new vendor dependencies. The [jQuery 3.x migration saga](https://core.trac.wordpress.org/ticket/37110) is a good example of the challenges this poses over time. React and Lodash are too pivotal and too large to _not_ expose, hence their inclusion. The same cannot necessarily be said about all utilities which Gutenberg uses.

**Comment by gziolo** [comment]
I tried to inline `rememo` in one of the existing modules in https://github.com/WordPress/gutenberg/pull/17863. I didn't notice huge benefits from doing it. It looks like this isn't a real issue for very small packages like `classnames`. See related comment from @senadir in https://github.com/WordPress/gutenberg/pull/17863#issuecomment-541577559:
> `createSelector` does make sense since selectors are used extensively in `wordpress/data`, but `classnames` doesn't make sense since it's already 300 bytes gziped

**Comment by gziolo** [comment]
Let’s close this issue for the time being as we decided that promoting smaller utility methods to public APIs would introduce too much work to maintain them. At the same the benefits like decreased bundles size are way too small. Let’s reopen this issue later when there are new ways to optimize the setup we have or we find bigger libraries to extract, similar to what we did with react, lodash or moment script handles or with inlining redux.
